Build failure: nginxModules.njs

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

Also happens on master, in the nixpkgs repo run:
```bash
nix-build -E 'with import ./default.nix {}; (nginx.override { modules = [nginxModules.njs]; })'
```

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/291280125

### Relevant log output

```console
gcc -c -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g   -I src/core -I src/event -I src/event/modules -I src/event/quic -I src/os/unix -I /nix/store/2ysp5ichpccf4lv1wp2qcwz0bmm840f1-rtmp -I /usr/include/libxml2 -I objs -I src/http -I src/http/modules -I src/http/v2 -I /build/nix/store/kc3hv49xzcnj84bjbyzg4p6z12zxsirv-hg-archive-nginx-njs/nginx/../src -I /build/nix/store/kc3hv49xzcnj84bjbyzg4p6z12zxsirv-hg-archive-nginx-njs/nginx/../build -I src/stream -I /build/nix/store/kc3hv49xzcnj84bjbyzg4p6z12zxsirv-hg-archive-nginx-njs/nginx/../src -I /build/nix/store/kc3hv49xzcnj84bjbyzg4p6z12zxsirv-hg-archive-nginx-njs/nginx/../build \
        -o objs/addon/external/njs_zlib_module.o \
        /build/nix/store/kc3hv49xzcnj84bjbyzg4p6z12zxsirv-hg-archive-nginx-njs/nginx/../external/njs_zlib_module.c
/build/nix/store/kc3hv49xzcnj84bjbyzg4p6z12zxsirv-hg-archive-nginx-njs/nginx/../external/njs_zlib_module.c:9:10: fatal error: zlib.h: No such file or directory
    9 | #include <zlib.h>
      |          ^~~~~~~~
compilation terminated.
make[1]: *** [objs/Makefile:2158: objs/addon/external/njs_zlib_module.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/build/nginx-1.26.3'
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.64, NixOS, 25.05 (Warbler), 25.05.20250213.2ff53fe`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `"nixos-23.05"`
 - nixpkgs: `/nix/store/r06xpql4r58fkhzq6np83zhpm7gdmwjv-22r7q7s9552gn1vpjigkbhfgcvhsrz68-source`

### Notify maintainers

@SuperSandro2000 
---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/servers/http/nginx/modules.nix
+++ b/pkgs/servers/http/nginx/modules.nix
@@ -508,8 +508,8 @@ let self = {
     src = fetchFromGitHub {
       owner = "nginx";
       repo = "njs";
-      rev = "0.8.7";
-      hash = "sha256-VEXzP+cN5hnDeniccwY7GIi4x460rnWO/o7ja3DyRCc=";
+      rev = "0.8.9";
+      hash = "sha256-TalS9EJP+vB1o3BKaTvXXnudjKhNOcob3kDAyeKej3c=";
     };
 
     # njs module sources have to be writable during nginx build, so we copy them

--- a/pkgs/servers/http/nginx/modules.nix
+++ b/pkgs/servers/http/nginx/modules.nix
@@ -1,5 +1,6 @@
 { lib
 , config
+, nixosTests
 , fetchFromGitHub
 , fetchFromGitLab
 , fetchhg
@@ -518,17 +519,19 @@ let self = {
       mkdir -p "$(dirname "$NJS_SOURCE_DIR")"
       cp --recursive "${src}" "$NJS_SOURCE_DIR"
       chmod -R u+rwX,go+rX "$NJS_SOURCE_DIR"
-      export configureFlags="''${configureFlags/"${src}"/"$NJS_SOURCE_DIR/nginx"}"
+      export configureFlags="''${configureFlags/"${src}"/"$NJS_SOURCE_DIR/nginx"} --with-ld-opt='-lz'"
       unset NJS_SOURCE_DIR
     '';
 
-    inputs = [ which ];
+    inputs = [ which zlib ];
+
+    passthru.tests = nixosTests.nginx-njs;
 
     meta = with lib; {
       description = "Subset of the JavaScript language that allows extending nginx functionality";
       homepage = "https://nginx.org/en/docs/njs/";
       license = with licenses; [ bsd2 ];
-      maintainers = [ ];
+      maintainers = with maintainers; [ jvanbruegge ];
     };
   };
 

